python/ambassador/envoy/v2/v2listener.py

```diff
@@ -512,380 +618,389 @@ def __init__(self, config: 'V2Config', listener: IRListener) -> None:
+            # We now have a secure route and an insecure route, so we need to walk all listeners
+            # and all vhosts, and match up the routes with the vhosts.
+
+            for port, listener in listeners_by_port.items():
+                for vhostname, vhost in listener.vhosts.items():
+                    # For each vhost, we need to look at things for the secure world as well
+                    # as the insecure world, depending on what the action is exactly (and note
+                    # that we can have an action of None if we're looking at a vhost created
+                    # by an insecure_addl_port).
+
+                    candidates = []
+
+                    if vhost._action is not None:
+                        candidates.append(( True, secure_route, vhost._action ))
+
+                    if vhost._insecure_action == "Redirect":
+                        candidates.append(( False, redirect_route, "Redirect" ))
+                    elif vhost._insecure_action is not None:
+                        candidates.append((False, insecure_route, vhost._insecure_action))
+
+                    for secure, route, action in candidates:
+                        variant = "secure" if secure else "insecure"
+
+                        if route["match"].get("prefix", None) == "/.well-known/acme-challenge/":
+                            # ACME challenges can never be secure.
+                            if secure:
+                                logger.info(
+                                    f"V2Listeners: {listener.name} {vhostname} secure: force Drop for ACME challenge")
+                                action = "Drop"
+                            else:
+                                # Force the action to "Route" for the insecure side of the world.
+                                logger.info(
+                                    f"V2Listeners: {listener.name} {vhostname} insecure: force Route for ACME challenge")
+                                action = "Route"
+                        elif route_hosts and (vhostname != '*') and (vhostname not in route_hosts):
+                            # Drop this because the host is mismatched.
+                            logger.info(
+                                f"V2Listeners: {listener.name} {vhostname} {variant}: Drop due to wrong host")
+                            action = "Drop"
+                        elif action == "Reject":
+                            logger.info(
+                                f"V2Listeners: {listener.name} {vhostname} {variant}: Drop due to Reject action")
+                            action = "Drop"
+                        else:
+                            logger.info(
+                                f"V2Listeners: {listener.name} {vhostname} {variant}: Accept as {action}")
+
+                        if action != 'Reject':
+                            vhost.append_route(route)
+                        # elif action == 'Redirect':
```

Comment:
Do we need this?

Reply:
It's actually about to come back (in a slightly modified fashion) for Edge Stack. So stand by. 🙂